ducktape/cluster/remoteaccount.py

```diff
@@ -62,6 +63,8 @@ def _can_ping_url(self, url, headers):
     def ssh_command(self, cmd):
         if self.local:
             return cmd
+        if self.sudo_user is not None:
```

Comment:
Are you suggesting to provide `sudo_user` in the `ssh_command(..)` instead of specifying it in the `RemoteAccount` constructor, so that ducktape user can specify `sudo_user` per ssh_command?

I have considered this as well. But I don't think any use case for this yet. In my opinion `RemoteAccount` allows user to run a `cmd` on a given `machine` (i.e. uri or ip) as the given `user`. This `sudo_user` is very similar to the existing `user`, which is not specified per ssh_command. I couldn't find a scenario where we will want to run `ssh_command` with different `sudo_user` value using a given `RemoteAccount` instance.

So my answer is no, we always want to apply the sudo user for a given `RemoteAccount` if it is specified in the constructor. What do you think?